src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetCultureCommand.cs

```diff
@@ -1,23 +1,61 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Globalization;
 using System.Management.Automation;
 
 namespace Microsoft.PowerShell.Commands
 {
     /// <summary>
     /// Returns the thread's current culture.
     /// </summary>
-    [Cmdlet(VerbsCommon.Get, "Culture", HelpUri = "https://go.microsoft.com/fwlink/?LinkID=113312")]
+    [Cmdlet(VerbsCommon.Get, "Culture", DefaultParameterSetName = NameParameterSet, HelpUri = "https://go.microsoft.com/fwlink/?LinkID=113312")]
     [OutputType(typeof(System.Globalization.CultureInfo))]
     public sealed class GetCultureCommand : PSCmdlet
     {
+        private const string NameParameterSet = "Name";
+        private const string ListAvailableParameterSet = "ListAvailable";
+
+        /// <summary>
+        /// Gets or sets the specified culture name.
+        /// </summary>
+        [Parameter(ParameterSetName = NameParameterSet)]
+        [ValidateNotNull]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets a switch to list all available cultures.
+        /// </summary>
+        [Parameter(ParameterSetName = ListAvailableParameterSet)]
+        public SwitchParameter ListAvailable { get; set; }
+
         /// <summary>
         /// Output the current Culture info object.
         /// </summary>
-        protected override void BeginProcessing()
+        protected override void EndProcessing()
```

Comment:
Is there a reason to do this in `EndProcessing` vs `ProcessRecord`?  Typically, `EndProcessing` is used for cleanup after the pipeline has completed.

Reply:
Initially we do not support pipelines and using EndProcessing() was ok. Currently the cmdlet was enhanced to support pipelines and the code was moved to ProcessRecord().